diff -up plymouth-0.7.0/scripts/plymouth-set-default-plugin.fix-set-default plymouth-0.7.0/scripts/plymouth-set-default-plugin
--- plymouth-0.7.0/scripts/plymouth-set-default-plugin.fix-set-default	2009-03-06 18:05:04.348305491 -0500
+++ plymouth-0.7.0/scripts/plymouth-set-default-plugin	2009-03-06 18:05:16.368307465 -0500
@@ -139,6 +139,6 @@ fi
 
 (cd ${LIBDIR}/plymouth;
         ln -sf ${PLUGIN_NAME}.so default.so && \
-        [ $DO_INITRD_REBUILD -ne 0 ] && \
-        $LIBEXECDIR/plymouth/plymouth-update-initrd)
+        ([ $DO_INITRD_REBUILD -ne 0 ] && \
+          $LIBEXECDIR/plymouth/plymouth-update-initrd) || :)
 
diff -up plymouth-0.7.0/scripts/plymouth-set-default-plugin.in.fix-set-default plymouth-0.7.0/scripts/plymouth-set-default-plugin.in
--- plymouth-0.7.0/scripts/plymouth-set-default-plugin.in.fix-set-default	2009-03-06 18:04:48.494054716 -0500
+++ plymouth-0.7.0/scripts/plymouth-set-default-plugin.in	2009-03-06 18:04:58.738052455 -0500
@@ -139,6 +139,6 @@ fi
 
 (cd ${LIBDIR}/plymouth;
         ln -sf ${PLUGIN_NAME}.so default.so && \
-        [ $DO_INITRD_REBUILD -ne 0 ] && \
-        $LIBEXECDIR/plymouth/plymouth-update-initrd)
+        ([ $DO_INITRD_REBUILD -ne 0 ] && \
+          $LIBEXECDIR/plymouth/plymouth-update-initrd) || :)
 
